### ISSUE TITLE
`weave start` with container name or short id leaks allocated IPs and DNS records

### ISSUE THREAD

**Issue by rade** [weaveworks/weave#881]
To reproduce
````
$ weave launch -iprange 10.3.0.0/24
c86dea6734a9bee5788e07a9cafa064178d5f930d4582e60d8d59ef7f7f3604b
$ weave launch-dns 10.3.1.1/24
6615e65710c480b30f492b03529b70313e7b6a89e31719f3274ab08d2b1e5328
$ docker create --name=foo -h foo.weave.local -ti ubuntu
ba6622c1301f614d5e38422324a718199629f45b6e22331b163e2878edd71b71
$ weave status | grep -E '(Free IPs|foo.weave.local)'
Allocator subnet 10.3.0.0/24  Free IPs: ~99.2%, 254 local, ~0 remote
$ weave start foo
foo
$ weave status | grep -E '(Free IPs|foo.weave.local)'
Allocator subnet 10.3.0.0/24  Free IPs: ~98.8%, 253 local, ~0 remote
foo: foo.weave.local.[10.3.0.1]
$ docker kill foo
foo
$ weave status | grep -E '(Free IPs|foo.weave.local)'
Allocator subnet 10.3.0.0/24  Free IPs: ~98.8%, 253 local, ~0 remote
foo: foo.weave.local.[10.3.0.1]
````

The problem stems from the fact that `docker start` just echoes back what it is given, rather than the full container id. Which is most useless.

This was [spotted by @bboreham](https://github.com/bboreham/weave/commit/064dfc6db0ce38db3dd733ad601250ee38cce782)

**Comment by rade** [comment]
The fix for this is trivial but we really should add some tests, at least for DNS (which should be easy; just check for a failing lookup). IPAM is harder.

**Comment by bboreham** [comment]
On the multi-subnet IPAM branch (#827), the fix is 064dfc6db0ce38db3dd733ad601250ee38cce782, the DNS test is c4f86fceb93cc6824aa692b965612f363317cef1 and the IPAM test is part of eb922759d5381720175b7794dce7afe252d0df0d
